app/src/main/java/org/coepi/android/ui/cen/CENViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package org.coepi.android.ui.cen
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import org.coepi.android.cen.CENRepo
+import org.coepi.android.cen.CEN
+import org.coepi.android.cen.CENReport
+import org.coepi.android.ui.navigation.RootNavigation
+
+class CENViewModel(repo: CENRepo,
+                   private val rootNav: RootNavigation) : ViewModel() {
+
+    // CEN being broadcast by this device
+    val myCurrentCEN = Transformations.map(repo.CEN) { cen ->
```

Comment:
We should use only RxJava transformations. The idea is to model the flows with RxJava and call `toLiveData()` only when interfacing with the UI at the very end ("outer edges").

We use `LiveData` only for the lifecycle management, which `RxJava` doesn't provide. This is purpose is fulfilled by using it like this.

In this case it would be (if you use RxJava in the repo):
```kotlin
repo.CEN.map { "CEN: $it"}.toLiveData()
```